app/components/application_complete_content_component.rb

```diff
@@ -0,0 +1,33 @@
+class ApplicationCompleteContentComponent < ActionView::Component::Base
+  include ViewHelper
+
+  validates :submitted_at, presence: true
+
+  def initialize(submitted_at:)
+    @submitted_at = submitted_at
+  end
+
+  def submitted_at_date
+    @submitted_at.strftime('%e %B %Y')
+  end
+
+  def respond_by_date
```

Comment:
The logic to calculate these dates probably needs to split out from the presentation logic, either in the `ApplicationForm` or a separate class. This will enable re-use and separate testing.

Reply:
Totally agree, I was thinking about this when building it – the business rules e.g. `+ n.days` shouldn't live here. The formatting e.g. `strftime` and adding `#{n} days` is fine imo.

Should we hoist it somewhere else now, or later?

Reply:
I would do it now!

Reply:
As discussed offline, will create a wrapping `ApplicationFormDates` model to hoist these.

Reply:
Done this, called it `ApplicationDates` since it seemed appropriate.